### ISSUE TITLE
add removeOnComplete option to Queue constructor

### ISSUE THREAD

**Issue by danprince** [OptimalBits/bull#842]
Following on from #840, it would be handy to be able to pass the `removeOnComplete` option when creating a queue, so that it could be applied to all calls to `Queue#add`.

```js
let jobs = new Queue({ settings: { removeOnComplete: true } });
jobs.add({ foo: "bar" });
jobs.add({ foo: "baz" });

// would be the same as

let jobs = new Queue({});
jobs.add({ foo: "bar" }, { removeOnComplete: true });
jobs.add({ foo: "baz" }, { removeOnComplete: true });
```

Although it might be more appropriate to call it `removeJobOnComplete` or `removeCompletedJobs`? So that it doesn't sound like the queue will be removed.

**Comment by mxstbr** [comment]
Fixed by #860 which adds a `defaultJobOptions` option to the Queue constructor. 

```JS
new Queue({
  defaultJobOptions: { removeOnComplete: true }
})
```

**Comment by marcospgp** [comment]
This parameter has no documentation associated - it shows up in the docs but is not explained anywhere. I only understood its functionality after finding this issue 😄
